src/components/Requirements.vue

```diff
@@ -1,214 +1,573 @@
-<template>
+<template v-if="semesters">
   <div class="requirements">
+    <div class="fixed">
     <h1 class="title">School Requirements</h1>
+
     <!-- loop through reqs array of req objects -->
-    <div class="reqs" v-for="(req, index) in reqs" :key="req.id">
+    <div class="req" v-for="(req, index) in reqs" :key="req.id">
       <div class="row top">
         <p class="name col p-0">{{ req.name }}</p>
         <div class="col-1 text-right p-0">
-          <button class="btn">
+          <button @click="modalShow = !modalShow" class="btn">
             <!-- svg for settings icon -->
             <img class="settings" src="../assets/images/gear.svg" />
           </button>
         </div>
       </div>
+
       <!-- progress bar settings -->
       <div class="progress">
         <div
           class="progress-bar"
-          :style="{
-            'background-color': req.color,
-            width: `${(req.progress / req.total) * 100}%`
-          }"
+          v-bind:style="{ 'background-color': `#${req.color}`, width: `${(req.fulfilled/req.required)*100}%`}"
           role="progressbar"
         ></div>
       </div>
+
       <p class="progress-text">
-        <strong>{{ req.progress }}/{{ req.total }}</strong> Total {{ req.count }} Inputted on
-        Schedule
+        <strong>{{ req.fulfilled }}/{{ req.required }}</strong>
+        Total {{ req.type }} Inputted on Schedule
       </p>
-      <!-- display button (off when req.displayDetails is false) -->
-      <button
-        class="view btn"
-        v-if="!req.displayDetails"
-        :style="{ 'background-color': req.color }"
-        @click="turnDetails(index, true)"
-      >
-        View All {{ req.type }} Requirements
-      </button>
-      <!-- when displayDetails is true -->
-      <div class="detail" v-if="req.displayDetails">
-        <div class="ongoing">
-          <div class="row detail-bar">
-            <p class="col detail-text p-0">In-Depth {{ req.type }} Requirements</p>
-            <div class="text-right p-0">
-              <button class="btn" @click="turnDetails(index, false)">
-                <!-- picture of close (x) icon -->
-                <img class="cancel" src="../assets/images/x.png" alt="X" />
+
+      <!--View more college requirements -->
+      <div class="row top">
+        <div class="col-1 p-0" >
+          <button v-bind:style="{ 'color': `#${req.color}` }" class="btn" v-on:click="toggleDetails(index)" style="color:#1AA9A5;">
+            <!-- svg for dropdown icon -->
+            <img
+              class="arrow"
+              style="fill: #1AA9A5; color:#1AA9A5 "
+              src="../assets/images/dropdown-blue.svg"
```

Comment:
It can sometimes be better to reference the home directory with "@" instead of using "../" (so "@/assets/images...") in case this Vue file is ever moved to a subfolder. Not a necessary change I don't think but just a future note.